controllers/tables.go

```diff
@@ -135,14 +136,26 @@ func SelectFromTables(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	selectFields := postgres.SelectByRequest(r)
-	if selectFields == "" {
-		selectFields = statements.SelectInTable
+	permission := postgres.TablePermissions(table, "read")
+	if !permission {
+		log.Println("You don't have permission for this action.")
+		http.Error(w, "Unable to parse table in URI", http.StatusMethodNotAllowed)
+		return
 	}
-	query := fmt.Sprintf("%s %s.%s.%s", selectFields, database, schema, table)
 
-	countQuery := postgres.CountByRequest(r)
+	// get selected columns, "*" if empty "_columns"
+	cols := postgres.SelectByRequest(r)
+	cols = postgres.FieldsPermissions(table, cols, "read")
+	if len(cols) == 0 {
+		log.Println("You don't have permission for this action. Please check the permitted fields for this table.")
+		http.Error(w, "You don't have permission for this action. Please check the permitted fields for this table.", http.StatusUnauthorized)
+		return
+	}
+	colsStr := strings.Join(cols, ",")
```

Comment:
In controllers is not place to create SQL operations, it's rule of `postgres` package.